module-apps/application-messages/windows/NewMessage.cpp

```diff
@@ -46,14 +46,40 @@ namespace gui
             state.reset();
         }
 
-      private:
+      protected:
         std::unique_ptr<gui::TextBackup> state;
     };
 
+    class NewMessageWindow::RecipientMemento : public NewMessageWindow::MessageMemento
+    {
+      public:
+        void setState(const gui::Text *_state, const std::shared_ptr<ContactRecord> _contact)
+        {
+            if (_contact != nullptr && !_contact->numbers.empty()) {
+                phoneNumberView = std::make_unique<utils::PhoneNumber::View>(_contact->numbers.at(0).number);
+            }
+
+            MessageMemento::setState(_state);
+        }
+
+        void restoreState(gui::Text *_state, utils::PhoneNumber::View *_phoneNumberView)
```

Comment:
same as above
what's the difference between _phoneNumberView and phoneNumberView
I would consider naming it as newPhoneNumber and currentPhoneNumber. If this is what this is about